graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/fetchers/deployed/DeployedFetcher.java

```diff
@@ -371,16 +368,49 @@ protected ResultSet executeUnchecked(
    *     argument).
    * @param getArgument how to get the value of an argument (same).
    */
-  protected List<BuiltCondition> bind(
-      List<WhereConditionModel> whereConditions,
+  protected List<BuiltCondition> bindWhere(
+      List<WhereConditionModel> conditions,
       EntityModel entity,
       Predicate<String> hasArgument,
       Function<String, Object> getArgument,
       Keyspace keyspace) {
 
     List<BuiltCondition> result = new ArrayList<>();
     List<WhereConditionModel> activeConditions = new ArrayList<>();
-    for (WhereConditionModel whereCondition : whereConditions) {
+    bind(conditions, hasArgument, getArgument, keyspace, result, activeConditions);
+
+    // Re-check the validity of the query (we already did while building the model, but some
+    // conditions might not apply if the corresponding argument was missing).
+    entity
+        .validateNoFiltering(activeConditions)
+        .ifPresent(
+            message -> {
+              throw new IllegalArgumentException("Invalid arguments: " + message);
+            });
+    return result;
+  }
+
+  protected List<BuiltCondition> bindIf(
+      List<IfConditionModel> conditions,
+      Predicate<String> hasArgument,
+      Function<String, Object> getArgument,
+      Keyspace keyspace) {
+
+    List<BuiltCondition> result = new ArrayList<>();
+    List<IfConditionModel> activeConditions = new ArrayList<>();
+    bind(conditions, hasArgument, getArgument, keyspace, result, activeConditions);
```

Comment:
We don't use `activeConditions` in this method (which is right BTW: removing IF conditions cannot make a request invalid).

Maybe we could adapt `bind()` to only build it when needed, but that would make it a bit more complicated. Creating one extra list is not the end of the world, I'm fine keeping it like this.